internal/types/types.go

```diff
@@ -1700,3 +1701,29 @@ type Webhook struct {
 	CreatedByUserID int32
 	UpdatedByUserID int32
 }
+
+// CodeHostURN is an identifier for a unique code host in the form
+// of its host URL.
+// To create a new CodeHostURN, use NewCodeHostURN.
+// e.g. NewCodeHostURN("https://github.com")
+// To use the string value again, use codeHostURN.String()
+type CodeHostURN struct {
+	urlString string
+}
+
+// NewCodeHostURN takes a code host URL string, normalizes it,
+// and returns a CodeHostURN. If the string is required, use the
+// .String() method on the CodeHostURN.
+func NewCodeHostURN(urnString string) (CodeHostURN, error) {
```

Comment:
```suggestion
func NewCodeHostURN(urn string) (CodeHostURN, error) {
```

Not suggesting everywhere else. I figured it would be easier in an editor to do a `find-and-replace`.